packages/css-to-js/test/api.test.js

```diff
@@ -168,6 +165,27 @@ describe("@modular-css/css-to-js API", () => {
         expect(namedExports).toEqual([ "$values", "b" ]);
     });
 
+
+    it("should generate javascript from composes", async () => {
+        const processor = new Processor({ resolvers });
+
+        await processor.file(require.resolve("./specimens/composes/custom-first-rule.css"));
+
+        const { code } = transform(require.resolve("./specimens/composes/custom-first-rule.css"), processor);
+
+        expect(code).toMatchSnapshot("code");
+    });
+
+    it("should generate javascript from multiple composes", async () => {
```

Comment:
will local composes declaration work, if referenced class is defined after usage?

Reply:
If you mean something like [this](https://m-css.com/repl/#NrBEHoFsEMEsDsB0kDGBnNoA0pEDMB7AgAgG8AdeY64lAyABwLQFM0AuYgI2gCcBuSgF9KlRD15lKNWgQA2BXp14sAJoPhDQAXW1A) then in such a case the `processor` throws
https://github.com/tivac/modular-css/blob/e64362e087f4ae6113a1b622e59dab827a9f2e97/packages/processor/plugins/composition.js#L66-L71

Reply:
When we switched to the new walking APIs in `postcss@8` it meant that out-of-order declarations would've been a bunch of extra work to support, so I chose not too.

Doesn't mean you couldn't support them using a "gather and process" approach, and for at least one type we do that, but given how `composes` can be several levels deep it'd require some real careful planning.